containers/ecr-viewer/src/app/services/evaluateFhirDataService.tsx

```diff
@@ -292,6 +293,139 @@ export const evaluateOccupation = (fhirBundle: Bundle) => {
+ * @param fhirBundle - The FHIR bundle containing pregnancy data.
+ * @returns An array of evaluated and formatted pregnancy data.
+ */
+export const evaluatePregnancyData = (fhirBundle: Bundle) => {
+  const lastMenstrualPeriodObservations = evaluateAll(
+    fhirBundle,
+    fhirPathMappings.lastMenstrualPeriod,
+  );
+  const pregnancyStatusObservations = evaluateAll(
+    fhirBundle,
+    fhirPathMappings.pregnancyStatus,
+  );
+  const postpartumStatusObservations = evaluateAll(
+    fhirBundle,
+    fhirPathMappings.postpartumStatus,
+  );
+  const allPregnancyObservations = [
+    ...lastMenstrualPeriodObservations.map((ob) => {
+      return { type: "Last Menstrual Period", observation: ob };
+    }),
+    ...pregnancyStatusObservations.map((ob) => {
+      return { type: "Pregnancy Status", observation: ob };
+    }),
+    ...postpartumStatusObservations.map((ob) => {
+      return { type: "Postpartum Status", observation: ob };
+    }),
+  ].sort(sortPregnancyObservations);
+
+  if (allPregnancyObservations.length === 0) return;
+
+  return (
+    <ExpandCollapseAccordion
+      className="accordion-rr"
+      descriptor="pregnancy info"
+      items={allPregnancyObservations.map((obs) => {
+        let data: DisplayDataProps[] = [];
+        const { type, observation } = obs;
+        if (type === "Last Menstrual Period") {
+          data = [
+            {
+              title: "Last Menstrual Period",
+              value: formatDate(
+                evaluateValue(observation, fhirPathMappings.effectiveX),
+              ),
+            },
+          ];
+        } else if (type === "Pregnancy Status") {
+          data = [
+            {
+              title: "Status",
+              value: evaluateValue(observation, "valueCodeableConcept"),
+            },
+            {
+              title: "Effective Date",
+              value: formatStartEndDateTime(
+                evaluateOne(observation, fhirPathMappings.effectivePeriod),
+              ),
+            },
+          ];
+        } else if (type === "Postpartum Status") {
+          data = [
+            {
+              title: "Status",
+              value: evaluateValue(observation, "valueCodeableConcept"),
+            },
+            {
+              title: "Effective Date",
+              value: formatDate(
+                evaluateValue(observation, fhirPathMappings.effectiveX),
+              ),
+            },
+          ];
+        }
+
+        const content = (
+          <>
+            {data.map((d) => {
+              return <DataDisplay item={d} />;
+            })}
+          </>
+        );
+
+        return {
+          title: (
+            <div className="display-flex flex-row flex-no-wrap flex-justify">
+              <span className="text-base">{type}</span>
+            </div>
+          ),
+          expanded: false,
+          content,
+          id: `${Math.random()}`,
```

Comment:
do the observations have id's we can leverage? (need the fallback still, but avoid leaning on it too much)